src/test/java/se/knowit/bookitregistration/controller/RegistrationControllerTest.java

```diff
@@ -0,0 +1,103 @@
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(MockitoExtension.class)
+class RegistrationControllerTest {
+    public static final String PATH = "/api/v1/registrations";
+    private static final UUID DEFAULT_UUID = UUID.fromString("72ab7c8b-c0d5-4ab2-8c63-5cf1ad0b439b");
+
+    @Mock
+    private RegistrationService registrationService;
+
+    @InjectMocks
+    private RegistrationController registrationController;
+
+    private MockMvc mockMvc;
+
+    private RegistrationMapper mapper = new RegistrationMapper();
+
+    @BeforeEach
+    void setUp() {
+        mockMvc = MockMvcBuilders.standaloneSetup(registrationController).build();
+    }
+
+    @Test
+    void postRequest_WithValidData_ShouldReturn_201Created() throws Exception {
+        String json = "{\"eventId\": \"" + UUID.randomUUID() + "\", \"email\": \"test@test.com\"}";
+        RegistrationDTO dto = getRegistrationDTOFromJson(json);
+        Registration savedRegistration = mapper.fromDTO(dto);
+        savedRegistration.setRegistrationId(DEFAULT_UUID);
+        savedRegistration.setId(1L);
+
+        when(registrationService.save(eq(mapper.fromDTO(dto)))).thenReturn(savedRegistration);
+
+        MvcResult result = mockMvc.perform(
+                post(PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(json))
+                .andExpect(status().isCreated())
+                .andReturn();
+    }
+
+    @Test
+    void postRequest_WithInvalidData_ShouldReturn_400() throws Exception {
+
+        when(registrationService.save(any())).thenThrow(IllegalArgumentException.class);
+
+        mockMvc.perform(
+                post(PATH)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content("{\"badRequest\" : \"true\"}"))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void postRequest_DuplicateRegistration_ShouldReturn_409() throws Exception {
+        String incomingJson = "{\"eventId\" : \"" + UUID.randomUUID().toString() + "\", \"email\" : \" + test@test.com\"}";
+        MockHttpServletRequestBuilder postRequest = post(PATH)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(incomingJson);
+        mockMvc.perform(postRequest).andReturn();
+        when(registrationService.save(any())).thenThrow(ConflictingEntityException.class);
+        mockMvc.perform(postRequest).andExpect(status().isConflict());
+    }
+
+    @Test
+    void postRequest_Delete_ShouldReturn_204() throws Exception {
+        mockMvc.perform(
+                delete(PATH + "/" + DEFAULT_UUID))
+                .andExpect(status().isNoContent());
+
+    }
```

Comment:
Could check if the resource was actually deleted or not, and have 2 tests covering each outcome.